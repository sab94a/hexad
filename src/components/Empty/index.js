// @flow 

import React from 'react';

const Empty = () => (
    <div>There are no any items..</div>
);

export default Empty;
