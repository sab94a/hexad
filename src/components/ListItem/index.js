import React from 'react';

const ListItem = ({ title }) => (<div>{ title }</div>);

export default React.memo(ListItem);
