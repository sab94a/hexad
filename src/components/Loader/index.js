// @flow

import React from 'react';
import CircularProgress from '@material-ui/core/CircularProgress';
import { makeStyles } from '@material-ui/core/styles';

const useStyles = makeStyles(theme => ({
  progress: {
    margin: theme.spacing(2),
  },
}));

const Loader = () => {
    const classes = useStyles();

    return (
        <CircularProgress className={classes.progress} />
    );
}

export default Loader;
