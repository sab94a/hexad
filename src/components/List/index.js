import React from 'react';
import Empty from 'components/Empty';
import Loader from 'components/Loader';
import ListItem from 'components/ListItem';

const List = ({ items, loading }) => {
    const isEmpty = !items.length;

    return (
        <div>
            {loading && (
                <Loader />
            )}

            {!loading && isEmpty && (
                <Empty />
            )}

            {!loading && !isEmpty && items.map(({ title, id }) => (
                <ListItem title={ title } key={ id } />
            ))}
        </div>
    );
};

export default React.memo(List);
