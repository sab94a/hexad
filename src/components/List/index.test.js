import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import ListItem from 'components/ListItem';
import Loader from 'components/Loader';
import Empty from 'components/Empty';
import List from './';

const ITEMS = [
    { id: 1, title: 'Title 1', image: 'src', rate: 1, description: 'test' },
    { id: 2, title: 'Title 2', image: 'src', rate: 1, description: 'test' },
    { id: 3, title: 'Title 3', image: 'src', rate: 1, description: 'test' }
];

describe('List', () => {
    let component;

    it('Should render correct amount of item', () => {
        component = shallow(<List items={ ITEMS }/>);

        expect(shallowToJson(component)).toMatchSnapshot();
        expect(component.find(ListItem)).toHaveLength(ITEMS.length);
    });

    it('Should render loading status insted of items', () => {
        component = shallow(<List items={ ITEMS } loading={ true } />);

        expect(shallowToJson(component)).toMatchSnapshot();
        expect(component.find(Loader)).toHaveLength(1);
        expect(component.find(ListItem)).toHaveLength(0);
    });

    it('Should render empty status insted of items if array is empty', () => {
        component = shallow(<List items={ [] } />);

        expect(shallowToJson(component)).toMatchSnapshot();
        expect(component.find(Empty)).toHaveLength(1);
        expect(component.find(ListItem)).toHaveLength(0);
    });
});
