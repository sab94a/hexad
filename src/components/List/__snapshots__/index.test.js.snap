// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List Should render correct amount of item 1`] = `
<div>
  <Memo(ListItem)
    key="1"
    title="Title 1"
  />
  <Memo(ListItem)
    key="2"
    title="Title 2"
  />
  <Memo(ListItem)
    key="3"
    title="Title 3"
  />
</div>
`;

exports[`List Should render empty status insted of items if array is empty 1`] = `
<div>
  <Empty />
</div>
`;

exports[`List Should render loading status insted of items 1`] = `
<div>
  <Loader />
</div>
`;
